### PR TITLE
[JUJU-1665] Clean up generated files

### PR DESCRIPTION
**Pull request by barrettj12**
## Changes
- A few of our generated files (mainly mocks) are out of sync - i.e. regenerating the mocks gives a different file to the one checked-in to VCS. I've regenerated these and committed the updated code.
- Some of the `go:generate` directives were outdated (referring to nonexistent types, old packages, etc), so I've gone through and fixed these.
- Add a GitHub Action to check that all generated code is in-sync.

## TODO
- [x] Fix certgen:
  - [x] Generated code is not `gofmt`-ed
  - [x] Doesn't obey the go generate convention where some line should match the regex:
    ```
    ^// Code generated .* DO NOT EDIT\.$
    ```
- [x] Check our other generation tools obey this convention.
- [x] CI should delete all gen'd files before regenerating - to check we don't have any crusty generated code hanging around.
- [x] Decide what to do with `testing/certs_generated.go`
- [x] clean up git history

## QA steps

Check that the new GitHub Action "Generate" passes.

### CODE REVIEW

File: .github/workflows/gen.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ steps.go-version.outputs.version }}
+        id: go
+
+      - name: go get dependencies
+        run: go get ./...
+
+      - name: Delete all mocks
```
Comment (hpidcock) [inline .github/workflows/gen.yml:50]: Is this so that mocks that are missing a go generate directive get flagged?

File: testing/certs.go [hunk 2]
```diff
@@ -8,7 +8,8 @@ import (
 	"time"
 )
 
-//go:generate go run github.com/juju/juju/generate/certgen
+// To regenerate the certificates in certs_generated, run:
```
Comment (hpidcock) [inline testing/certs.go:11]: :+1:

**Review by hpidcock — APPROVED** [review]
Awesome work, LGTM

File: .github/workflows/gen.yml (hunk 1 above)
Comment (barrettj12) [reply .github/workflows/gen.yml:50]: Yep that's right. Pretty sure we had some that should no longer be there

**Pull request merged**